File: frontend/src/components/LuxuryWelcome.js
import React, { useState } from 'react';
import './styles.css';

const LuxuryWelcome = ({ onAuthSuccess }) => {
  const [isSignIn, setIsSignIn] = useState(true);
  const [form, setForm] = useState({ email: '', password: '' });
  const [error, setError] = useState('');

  const handleChange = (e) => {
    setForm({ ...form, [e.target.name]: e.target.value });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    // Simulate authentication (replace with real API call)
    if (form.email && form.password) {
      setError('');
      onAuthSuccess();
    } else {
      setError('Please enter both email and password.');
    }
  };  return (
    <div className="min-h-screen flex items-center justify-center" style={{ background: 'var(--color-background)' }}>
      <div className="absolute inset-0" style={{ 
        background: 'radial-gradient(circle at top right, rgba(30, 64, 175, 0.15), transparent 60%), radial-gradient(circle at bottom left, rgba(191, 160, 84, 0.1), transparent 40%)'
      }}></div>
      
      <div className="container max-w-md relative z-10 mx-4">
        <div className="text-center mb-10">
          <h1 className="text-5xl font-bold mb-2" style={{ 
            background: 'var(--gradient-gold)', 
            WebkitBackgroundClip: 'text', 
            WebkitTextFillColor: 'transparent' 
          }}>DRIFT</h1>
          <p style={{ color: 'var(--color-text-secondary)' }} className="text-sm tracking-widest uppercase">
            Luxury Travel Experience
          </p>
        </div>
        
        <div className="p-8 rounded-2xl shadow-lg" style={{ 
          background: 'var(--color-surface)',
          border: '1px solid rgba(255,255,255,0.05)',
          boxShadow: 'var(--shadow-lg)'
        }}>
          <div className="mb-6">
            <h2 className="text-2xl font-semibold mb-2" style={{ color: 'var(--color-text)' }}>
              {isSignIn ? 'Welcome Back' : 'Create Account'}
            </h2>
            <p style={{ color: 'var(--color-text-secondary)' }}>
              {isSignIn 
                ? 'Sign in to continue your luxury travel experience' 
                : 'Join DRIFT to begin your personalized travel journey'}
            </p>
          </div>
          
          <form onSubmit={handleSubmit} className="flex flex-col gap-5">
            <div className="flex flex-col gap-1">
              <label htmlFor="email" className="text-sm font-medium" style={{ color: 'var(--color-text-secondary)' }}>Email</label>
              <input
                id="email"
                type="email"
                name="email"
                value={form.email}
                onChange={handleChange}
                className="p-3 rounded-lg border-0 focus:ring-2 focus:ring-accent/50 transition-all"
                style={{ 
                  background: 'rgba(30, 30, 30, 0.7)', 
                  color: 'var(--color-text)',
                  boxShadow: 'inset 0 2px 4px rgba(0,0,0,0.3)' 
                }}
                required
              />
            </div>
            
            <div className="flex flex-col gap-1">
              <label htmlFor="password" className="text-sm font-medium" style={{ color: 'var(--color-text-secondary)' }}>Password</label>
              <input
                id="password"
                type="password"
                name="password"
                value={form.password}
                onChange={handleChange}
                className="p-3 rounded-lg border-0 focus:ring-2 focus:ring-accent/50 transition-all"
                style={{ 
                  background: 'rgba(30, 30, 30, 0.7)', 
                  color: 'var(--color-text)',
                  boxShadow: 'inset 0 2px 4px rgba(0,0,0,0.3)' 
                }}
                required
              />
            </div>
            
            {error && (
              <div className="bg-red-900/20 px-3 py-2 rounded text-sm" style={{ color: '#FFA0A0' }}>
                {error}
              </div>
            )}
            
            <button
              type="submit"
              className="py-3 px-4 rounded-lg mt-4 text-center font-semibold transition-all transform hover:scale-[1.02] active:scale-[0.98] flex items-center justify-center"
              style={{ 
                background: 'var(--gradient-gold)', 
                color: '#121212', 
                boxShadow: 'var(--shadow-md)' 
              }}
            >
              {isSignIn ? 'Sign In' : 'Create Account'}
            </button>
          </form>
          
          <div className="mt-8 pt-6 text-center" style={{ borderTop: '1px solid rgba(255,255,255,0.05)' }}>
            {isSignIn ? (
              <div>
                <p style={{ color: 'var(--color-text-secondary)' }}>New to DRIFT?</p>
                <button 
                  onClick={() => setIsSignIn(false)}
                  className="mt-2 font-medium hover:underline transition-all"
                  style={{ color: 'var(--color-accent)' }}
                >
                  Create an account
                </button>
              </div>
            ) : (
              <div>
                <p style={{ color: 'var(--color-text-secondary)' }}>Already have an account?</p>
                <button 
                  onClick={() => setIsSignIn(true)}
                  className="mt-2 font-medium hover:underline transition-all"
                  style={{ color: 'var(--color-accent)' }}
                >
                  Sign in
                </button>
              </div>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};

export default LuxuryWelcome;

File: flutter_app/lib/screens/luxury_welcome.dart
import 'package:flutter/material.dart';
import 'package:flutter_app/theme.dart';
import 'package:shared_preferences/shared_preferences.dart';
import 'package:flutter_secure_storage/flutter_secure_storage.dart';

class LuxuryWelcome extends StatefulWidget {
  final Function onAuthSuccess;
  
  const LuxuryWelcome({super.key, required this.onAuthSuccess});

  @override
  State<LuxuryWelcome> createState() => _LuxuryWelcomeState();
}

class _LuxuryWelcomeState extends State<LuxuryWelcome> {
  bool isSignIn = true;
  final TextEditingController emailController = TextEditingController();
  final TextEditingController passwordController = TextEditingController();
  String? error;
  bool rememberMe = false;
  final FlutterSecureStorage _secureStorage = const FlutterSecureStorage();
  bool _isLoading = true;

  @override
  void initState() {
    super.initState();
    _loadSavedCredentials();
  }

  Future<void> _loadSavedCredentials() async {
    try {
      final prefs = await SharedPreferences.getInstance();
      final shouldRemember = prefs.getBool('rememberMe') ?? false;
      
      if (shouldRemember) {
        final email = await _secureStorage.read(key: 'saved_email');
        final password = await _secureStorage.read(key: 'saved_password');
        
        if (email != null && password != null) {
          setState(() {
            emailController.text = email;
            passwordController.text = password;
            rememberMe = true;
          });
        }
      }
    } catch (e) {
      // Handle any errors silently
      debugPrint('Error loading saved credentials: $e');
    } finally {
      setState(() {
        _isLoading = false;
      });
    }
  }

  Future<void> _saveCredentials() async {
    try {
      final prefs = await SharedPreferences.getInstance();
      await prefs.setBool('rememberMe', rememberMe);
      
      if (rememberMe) {
        await _secureStorage.write(
          key: 'saved_email',
          value: emailController.text,
        );
        await _secureStorage.write(
          key: 'saved_password',
          value: passwordController.text,
        );
      } else {
        await _secureStorage.delete(key: 'saved_email');
        await _secureStorage.delete(key: 'saved_password');
      }
    } catch (e) {
      debugPrint('Error saving credentials: $e');
    }
  }

  @override
  void dispose() {
    emailController.dispose();
    passwordController.dispose();
    super.dispose();
  }

  Future<void> _handleSubmit() async {
    if (emailController.text.isEmpty || passwordController.text.isEmpty) {
      setState(() {
        error = 'Please enter both email and password.';
      });
      return;
    }
    
    setState(() {
      error = null;
    });
    
    // Save credentials if "Remember me" is checked
    await _saveCredentials();
    
    // Simulate authentication (just like in the web version)
    widget.onAuthSuccess();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Container(
        decoration: const BoxDecoration(
          gradient: DriftTheme.backgroundGradient,
        ),
        child: _isLoading 
          ? const Center(
              child: CircularProgressIndicator(
                valueColor: AlwaysStoppedAnimation<Color>(DriftTheme.gold),
              ),
            )
          : Stack(
          children: [
            // Background decorative elements
            Positioned(
              top: 0,
              right: 0,
              child: Container(
                width: 300,
                height: 300,
                decoration: BoxDecoration(
                  gradient: RadialGradient(
                    colors: [DriftTheme.gold.withAlpha(38), Colors.transparent],
                    radius: 1.0,
                  ),
                ),
              ),
            ),
            Positioned(
              bottom: 0,
              left: 0,
              child: Container(
                width: 200,
                height: 200,
                decoration: BoxDecoration(
                  gradient: RadialGradient(
                    colors: [DriftTheme.gold.withAlpha(25), Colors.transparent],
                    radius: 1.0,
                  ),
                ),
              ),
            ),
            
            // Main content
            Center(
              child: SingleChildScrollView(
                child: Padding(
                  padding: const EdgeInsets.all(24.0),
                  child: Column(
                    mainAxisSize: MainAxisSize.min,
                    children: [
                      // Logo and title
                      ShaderMask(
                        shaderCallback: (bounds) => DriftTheme.goldGradient.createShader(bounds),
                        child: const Text(
                          'DRIFT',
                          style: TextStyle(
                            fontSize: 48,
                            fontWeight: FontWeight.bold,
                            color: Colors.white,
                          ),
                        ),
                      ),
                      const SizedBox(height: 8),
                      const Text(
                        'LUXURY TRAVEL EXPERIENCE',
                        style: TextStyle(
                          fontSize: 12,
                          letterSpacing: 2,
                          color: DriftTheme.textSecondary,
                        ),
                      ),
                      const SizedBox(height: 40),
                      
                      // Auth Card
                      Container(
                        width: 400,
                        constraints: const BoxConstraints(maxWidth: 400),
                        decoration: BoxDecoration(
                          color: DriftTheme.surface,
                          borderRadius: BorderRadius.circular(16),
                          border: Border.all(
                            color: Colors.white.withAlpha(13),
                          ),
                          boxShadow: [
                            BoxShadow(
                              color: Colors.black.withAlpha(77),
                              blurRadius: 20,
                              spreadRadius: 2,
                            ),
                          ],
                        ),
                        child: Padding(
                          padding: const EdgeInsets.all(24.0),
                          child: Column(
                            crossAxisAlignment: CrossAxisAlignment.start,
                            children: [
                              Text(
                                isSignIn ? 'Welcome Back' : 'Create Account',
                                style: Theme.of(context).textTheme.headlineSmall,
                              ),
                              const SizedBox(height: 8),
                              Text(
                                isSignIn 
                                  ? 'Sign in to continue your luxury travel experience'
                                  : 'Join DRIFT to begin your personalized travel journey',
                                style: const TextStyle(color: DriftTheme.textSecondary),
                              ),
                              const SizedBox(height: 24),
                              
                              // Email Field
                              const Text(
                                'Email',
                                style: TextStyle(
                                  fontSize: 14, 
                                  color: DriftTheme.textSecondary,
                                  fontWeight: FontWeight.w500,
                                ),
                              ),
                              const SizedBox(height: 8),
                              TextField(
                                controller: emailController,
                                decoration: const InputDecoration(
                                  hintText: 'Enter your email',
                                ),
                                style: const TextStyle(color: DriftTheme.textPrimary),
                                keyboardType: TextInputType.emailAddress,
                              ),
                              const SizedBox(height: 16),
                              
                              // Password Field
                              const Text(
                                'Password',
                                style: TextStyle(
                                  fontSize: 14, 
                                  color: DriftTheme.textSecondary,
                                  fontWeight: FontWeight.w500,
                                ),
                              ),
                              const SizedBox(height: 8),
                              TextField(
                                controller: passwordController,
                                decoration: const InputDecoration(
                                  hintText: 'Enter your password',
                                ),
                                style: const TextStyle(color: DriftTheme.textPrimary),
                                obscureText: true,
                              ),
                              const SizedBox(height: 16),
                              
                              // Remember me checkbox
                              if (isSignIn)
                                Row(
                                  children: [
                                    SizedBox(
                                      width: 24,
                                      height: 24,
                                      child: Checkbox(
                                        value: rememberMe,
                                        activeColor: DriftTheme.gold,
                                        checkColor: DriftTheme.background,
                                        side: BorderSide(color: DriftTheme.goldLight.withAlpha(128)),
                                        shape: RoundedRectangleBorder(
                                          borderRadius: BorderRadius.circular(4),
                                        ),
                                        onChanged: (value) {
                                          setState(() {
                                            rememberMe = value ?? false;
                                          });
                                        },
                                      ),
                                    ),
                                    const SizedBox(width: 8),
                                    GestureDetector(
                                      onTap: () {
                                        setState(() {
                                          rememberMe = !rememberMe;
                                        });
                                      },
                                      child: const Text(
                                        'Remember me',
                                        style: TextStyle(
                                          color: DriftTheme.textSecondary,
                                          fontSize: 14,
                                        ),
                                      ),
                                    ),
                                  ],
                                ),
                              const SizedBox(height: 16),
                              
                              // Error message
                              if (error != null)
                                Container(
                                  padding: const EdgeInsets.all(12),
                                  decoration: BoxDecoration(
                                    color: Colors.red.withAlpha(25),
                                    borderRadius: BorderRadius.circular(8),
                                  ),
                                  child: Text(
                                    error!,
                                    style: const TextStyle(color: Colors.red),
                                  ),
                                ),
                              const SizedBox(height: 24),
                              
                              // Sign In/Create Account Button
                              SizedBox(
                                width: double.infinity,
                                child: ElevatedButton(
                                  onPressed: _handleSubmit,
                                  style: ElevatedButton.styleFrom(
                                    padding: const EdgeInsets.symmetric(vertical: 14),
                                    backgroundColor: DriftTheme.gold,
                                    shape: RoundedRectangleBorder(
                                      borderRadius: BorderRadius.circular(8),
                                    ),
                                  ),
                                  child: Text(
                                    isSignIn ? 'Sign In' : 'Create Account',
                                    style: const TextStyle(
                                      color: Colors.black87,
                                      fontWeight: FontWeight.bold,
                                    ),
                                  ),
                                ),
                              ),
                              const SizedBox(height: 24),
                              
                              // Toggle between sign in and sign up
                              Divider(color: Colors.white.withAlpha(25)),
                              const SizedBox(height: 16),
                              Center(
                                child: Column(
                                  children: [
                                    Text(
                                      isSignIn ? 'New to DRIFT?' : 'Already have an account?',
                                      style: const TextStyle(color: DriftTheme.textSecondary),
                                    ),
                                    const SizedBox(height: 8),
                                    TextButton(
                                      onPressed: () {
                                        setState(() {
                                          isSignIn = !isSignIn;
                                          error = null;
                                          // Clear form when switching to sign up
                                          if (!isSignIn) {
                                            emailController.clear();
                                            passwordController.clear();
                                          }
                                        });
                                      },
                                      child: Text(
                                        isSignIn ? 'Create an account' : 'Sign in',
                                        style: const TextStyle(color: DriftTheme.goldLight),
                                      ),
                                    ),
                                  ],
                                ),
                              ),
                            ],
                          ),
                        ),
                      ),
                    ],
                  ),
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
